conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/WebSecurityHandlerTest.java

```diff
@@ -51,48 +48,49 @@ public static void afterClass() {
+    public void testInternetExplorer11() {
         testFallbackSecurityPolicyHeader("Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko");
     }
 
-    private void testFallbackSecurityPolicyHeader(String userAgent) throws IOException {
-        try (Response response = client.newCall(new Request.Builder()
-                        .get()
-                        .url("http://localhost:12345")
-                        .header(HttpHeaders.USER_AGENT, userAgent)
-                        .build())
-                .execute()) {
-            validateCommonResponseHeaders(response);
-            assertThat(response.header("X-Content-Security-Policy"))
-                    .isEqualTo("default-src 'self'; img-src 'self' data:; "
-                            + "style-src 'self' 'unsafe-inline'; frame-ancestors 'self';");
-        }
+    private void testFallbackSecurityPolicyHeader(String userAgent) {
+        HttpURLConnection connection = openConnectionToTestServer();
+        connection.setRequestProperty(HttpHeaders.USER_AGENT, userAgent);
+        validateCommonResponseHeaders(connection);
+        assertThat(connection.getHeaderField("X-Content-Security-Policy"))
+                .isEqualTo("default-src 'self'; img-src 'self' data:; "
+                        + "style-src 'self' 'unsafe-inline'; frame-ancestors 'self';");
     }
 
-    private void validateCommonResponseHeaders(Response response) {
-        assertThat(response.header(HttpHeaders.CONTENT_SECURITY_POLICY))
+    private void validateCommonResponseHeaders(HttpURLConnection connection) {
+        assertThat(connection.getHeaderField(HttpHeaders.CONTENT_SECURITY_POLICY))
                 .isEqualTo("default-src 'self'; img-src 'self' data:; "
                         + "style-src 'self' 'unsafe-inline'; frame-ancestors 'self';");
-        assertThat(response.header(HttpHeaders.REFERRER_POLICY)).isEqualTo("strict-origin-when-cross-origin");
-        assertThat(response.header(HttpHeaders.X_CONTENT_TYPE_OPTIONS)).isEqualTo("nosniff");
-        assertThat(response.header(HttpHeaders.X_FRAME_OPTIONS)).isEqualTo("sameorigin");
-        assertThat(response.header(HttpHeaders.X_XSS_PROTECTION)).isEqualTo("1; mode=block");
+        assertThat(connection.getHeaderField(HttpHeaders.REFERRER_POLICY)).isEqualTo("strict-origin-when-cross-origin");
+        assertThat(connection.getHeaderField(HttpHeaders.X_CONTENT_TYPE_OPTIONS))
+                .isEqualTo("nosniff");
+        assertThat(connection.getHeaderField(HttpHeaders.X_FRAME_OPTIONS)).isEqualTo("sameorigin");
+        assertThat(connection.getHeaderField(HttpHeaders.X_XSS_PROTECTION)).isEqualTo("1; mode=block");
+    }
+
+    private static HttpURLConnection openConnectionToTestServer() {
+        try {
+            URL url = new URL("http://localhost:12345");
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
```

Comment:
GET is already the default, but listing it does make the behavior more obvious :+1:

Reply:
I will leave it then, better to be clear.